kernel/src/syscall/mount.rs

```diff
@@ -227,10 +229,16 @@ fn get_fs(
 
     let disk = if fs_type.properties().contains(FsProperties::NEED_DISK) {
         let devname = user_space.read_cstring(src_name_addr, MAX_FILENAME_LEN)?;
-        Some(
-            aster_block::get_device(devname.to_str().unwrap())
-                .ok_or(Error::with_message(Errno::ENOENT, "device does not exist"))?,
-        )
+        let path = devname.to_string_lossy();
+        let fs_path = FsPath::from_fd_and_path(AT_FDCWD, path.as_ref())?;
+        let path = ctx
+            .thread_local
+            .borrow_fs()
+            .resolver()
+            .read()
+            .lookup_no_follow(&fs_path)?;
+        let id = DeviceId::from_encoded_u64(path.metadata().rdev);
+        aster_block::lookup(id)
```

Comment:
If the FS type needs a backing block device but the lookup fails, then the mount operation will definitely fail. So I suppose that a proper error should be reported here.